packages/backend-app-api/src/services/implementations/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -98,8 +98,9 @@ export const rootHttpRouterServiceFactory = createServiceFactory(
       );
 
       lifecycle.addShutdownHook(() => server.stop());
-
-      await server.start();
+      lifecycle.addStartupHook(async () => {
+        await server.start();
```

Comment:
Do you recall why this was added? Seems to be counterproductive to the other ones - since there's only a single lifecycle events chain, and you don't know what order they get run in, others may be unblocked by the exact same event and start to make requests to a server that hasn't been started quite yet.

This, if anything, seems like a thing we may want to start eagerly right?

Reply:
Yeah, maybe you are right about this. But for the search at least missing the hook causes 500 errors from the catalogCollator and other collators depending on the HTTP server